### DISCUSSION TITLE
Expect syntax errors if feature not supported in target version

### DISCUSSION THREAD

**Discussion by s-cork** [astral-sh/ruff#10867] [Q&A] [answered]
Should the linter show syntax errors if the code is using syntax features not supported in the target version?

e.g.

```
inputs = []
while (data := input()) != 'exit':
    processed_data = data.upper()
    inputs.append(processed_data)

print("Final data:", inputs)

```

If my target is `py37` the walrus operator is not supported.
So it would be useful if I got an syntax error at that place in the code

**Comment by charliermarsh** [comment, marked as answer]
We don't do this today, but we want to. You can track it here: #6591.
